spec/rspec/core/profiler_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'rspec/core/profiler'
+
+RSpec.describe 'RSpec::Core::Profiler' do
+  let(:profiler) { RSpec::Core::Profiler.new }
+
+  it 'has starts with an empty hash of example_groups' do
+    expect(profiler.example_groups).to be_empty.and be_a Hash
+  end
+
+  context 'when hooked into the reporter' do
+    include FormatterSupport
+
+    let(:id) { 42 }
+    let(:description ) { "My Group" }
+    let(:now) { ::Time.now }
+
+    before do
+      allow(::RSpec::Core::Time).to receive(:now) { now }
+    end
+
+    def group
+      @group ||=
+        begin
+          group = super
+          allow(group).to receive_messages(:id => id, :top_level_description => description)
+          group
+        end
+    end
+
+    describe '#example_group_started' do
+      it 'records example groups start time and description via id' do
+        expect {
+          profiler.example_group_started group_notification group
+        }.to change { profiler.example_groups[id] }.to include(
+          :start => now, :description => description
+        )
```

Comment:
These `change { }.to include` expectations are prone to the issue raised in rspec/rspec-expectations#617: it'll pass as long as the hash changes somehow and includes these entries at the end, even if it already included these entries at the start.  I think it would be better to do something like:

``` ruby
to change { }.
  from(a_hash_excluding(:start, :description).
  to(a_hash_including(:start => now, :description => description)
```

That way it's explicit and avoids that issue.